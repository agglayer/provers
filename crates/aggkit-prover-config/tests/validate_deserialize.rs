use std::path::Path;

use aggkit_prover_config::ProverConfig as Config;
use insta::assert_toml_snapshot;
use pretty_assertions::assert_eq;

#[test]
fn empty_rpcs() {
    let input = "./tests/fixtures/validate_config/empty_rpcs.toml";

    let config = Config::try_load(Path::new(input)).unwrap();

    assert_toml_snapshot!(config, {
        ".storage.*" => insta::dynamic_redaction(|value, path| {
            if path.to_string() != "storage.db-path" {
                if let insta::internals::Content::String(path) = value {
                    return insta::internals::Content::String(path.replace(Path::new("./").canonicalize().unwrap().to_str().unwrap(), "/tmp/agglayer"));
                }
            }

            value
        }),
    });
}

#[test]
fn prover_grpc_max_decoding_message_size() {
    let input = "./tests/fixtures/validate_config/prover_grpc_max_decoding_message_size.toml";

    let config: Config = toml::from_str(&std::fs::read_to_string(input).unwrap()).unwrap();

    assert_toml_snapshot!(config, {
        ".storage.*" => insta::dynamic_redaction(|value, path| {
            if path.to_string() != "storage.db-path" {
                if let insta::internals::Content::String(path) = value {
                    return insta::internals::Content::String(path.replace(Path::new("./").canonicalize().unwrap().to_str().unwrap(), "/tmp/agglayer"));
                }
            }

            value
        }),
    });

    assert_eq!(config.grpc.max_decoding_message_size, 100 * 1024 * 1024);
}
