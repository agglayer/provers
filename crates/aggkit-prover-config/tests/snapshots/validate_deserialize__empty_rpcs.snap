---
source: crates/aggkit-prover-config/tests/validate_deserialize.rs
expression: config
snapshot_kind: text
---
grpc-endpoint = "127.0.0.1:8081"

[log]
level = "info"
outputs = []
format = "pretty"

[telemetry]
prometheus-addr = "0.0.0.0:3001"

[shutdown]
runtime-timeout = "30s"

[aggchain-proof-service.aggchain-proof-builder]
network-id = 0
proving-timeout = "1h"

[aggchain-proof-service.aggchain-proof-builder.primary-prover.network-prover]
proving-timeout = "5m"
sp1-cluster-endpoint = "https://rpc.production.succinct.xyz/"

[aggchain-proof-service.aggchain-proof-builder.contracts]
l1-rpc-endpoint = "http://anvil-mock-l1-rpc:8545/"
l2-execution-layer-rpc-endpoint = "http://anvil-mock-l2-rpc:8545/"
l2-consensus-layer-rpc-endpoint = "http://rollup-node-mock-l2-rpc:8545/"
polygon-rollup-manager = "0xb7f8bc63bbcad18155201308c8f3540b07f84f5e"
global-exit-root-manager-v2-sovereign-chain = "0xa40d5f56745a118d0906a34e69aec8c0db1cb8fa"
static-call-caller-address = "0x39027d57969ad59161365e0bbd53d2f63ee5aaa6"
evm-sketch-genesis = "mainnet"

[aggchain-proof-service.proposer-service]
mock = false
l1-rpc-endpoint = "http://anvil-mock-l1-rpc:8545/"

[aggchain-proof-service.proposer-service.client]
proposer-endpoint = "http://proposer-mock-rpc:3000/"
sp1-cluster-endpoint = "https://rpc.production.succinct.xyz/"
request-timeout = 600
proving-timeout = 3600

[primary-prover.network-prover]
proving-timeout = "5m"
sp1-cluster-endpoint = "https://rpc.production.succinct.xyz/"
